src/main/kotlin/app/revanced/extensions/Extensions.kt

```diff
@@ -27,6 +29,30 @@ internal fun MutableMethodImplementation.injectHideCall(
     )
 }
 
+/**
+ * traverse the class hierarchy starting from the given root class
+ *
+ * @param targetClass the class to start traversing the class hierarchy from
+ * @param callback function that is called for every class in the hierarchy
+ */
+fun BytecodeData.traverseClassHierarchy(targetClass: MutableClass, callback: MutableClass.() -> Unit) {
+    callback(targetClass)
+    this.findClass(targetClass.superclass ?: "")?.resolve()?.let {
```

Comment:
Is resolving the proxy required? Resolve creates a mutable clone of ImmutableClass. You should rely on the immutable type as long as you do not need mutability.

Reply:
idea behind this is that, if you're traversing the class hierarchy, you'll probably also want to mutate the classes in some way. 

Changing the function to 
```kotlin
fun BytecodeData.traverseClassHierarchy(targetClass: ClassDef, callback: ClassDef.() -> Unit) {
    targetClass.callback()
    this.findClass(targetClass.superclass ?: "")?.immutableClass?.let {
        traverseClassHierarchy(it, callback)
    }
}
```
and resolving in the callback does not work, as the changes done to the mutable class (superclass is changed) are not reflected in `immutableClass`.

Reply:
@Sculas, please check if this is really necessary.

Reply:
@Sculas Please also review this

Reply:
I'll look into it.